go/serialization.go

```diff
@@ -350,31 +350,59 @@ func SerializeSCTSignatureInput(sct SignedCertificateTimestamp, entry LogEntry)
 	}
 }
 
-func serializeV1SCT(sct SignedCertificateTimestamp) ([]byte, error) {
-	if err := checkExtensionsFormat(sct.Extensions); err != nil {
-		return nil, err
-	}
-	var buf bytes.Buffer
-	if err := binary.Write(&buf, binary.BigEndian, V1); err != nil {
-		return nil, err
-	}
-	if err := binary.Write(&buf, binary.BigEndian, sct.LogID); err != nil {
-		return nil, err
+func serializeV1SCTHere(sct SignedCertificateTimestamp, here []byte) (int, error) {
```

Comment:
How about returning `([]byte, error)`?
That way you can slice the buffer to the actual used length for the caller (they can always ignore that if it's not helpful for their use case), and if they pass in `nil` then you can allocate the right size, fill it and return - avoiding the slightly strange double call to this method below.

TBH though, I'm wondering whether all of this serialization section mightn't be made better/more idiomatic by being redone around `io.Writer`.